vscode-cairo/src/extension.ts

```diff
@@ -1,18 +1,21 @@
 import * as vscode from "vscode";
 import * as lc from "vscode-languageclient/node";
 import { setupLanguageServer } from "./cairols";
+import { Context } from "./context";
 
 let client: lc.LanguageClient | undefined;
 
-export async function activate(context: vscode.ExtensionContext) {
-  const config = vscode.workspace.getConfiguration();
-  const outputChannel = vscode.window.createOutputChannel("Cairo extension");
-  context.subscriptions.push(outputChannel);
+export async function activate(extensionContext: vscode.ExtensionContext) {
+  const ctx = Context.create(extensionContext);
 
-  if (config.get<boolean>("cairo1.enableLanguageServer")) {
-    client = await setupLanguageServer(config, context, outputChannel);
+  if (ctx.config.get<boolean>("enableLanguageServer")) {
```

Comment:
It'd be nice to keep the configs somewhere as consts and behind some typed facade